src/modules/peek/Peek.FilePreviewer/Previewers/ImagePreviewer/Helpers/PropertyHelper.cs

```diff
@@ -16,14 +17,21 @@ public static Task<Size> GetImageSize(string filePath)
         {
             return Task.Run(() =>
             {
-                var propertyStore = PropertyStoreShellApi.GetPropertyStoreFromPath(filePath, PropertyStoreShellApi.PropertyStoreFlags.OPENSLOWITEM);
-                if (propertyStore != null)
+                try
                 {
-                    var width = (int)PropertyStoreShellApi.GetUIntFromPropertyStore(propertyStore, PropertyStoreShellApi.PropertyKey.ImageHorizontalSize);
-                    var height = (int)PropertyStoreShellApi.GetUIntFromPropertyStore(propertyStore, PropertyStoreShellApi.PropertyKey.ImageVerticalSize);
+                    var propertyStore = PropertyStoreShellApi.GetPropertyStoreFromPath(filePath, PropertyStoreShellApi.PropertyStoreFlags.OPENSLOWITEM);
+                    if (propertyStore != null)
+                    {
+                        var width = (int)PropertyStoreShellApi.GetUIntFromPropertyStore(propertyStore, PropertyStoreShellApi.PropertyKey.ImageHorizontalSize);
+                        var height = (int)PropertyStoreShellApi.GetUIntFromPropertyStore(propertyStore, PropertyStoreShellApi.PropertyKey.ImageVerticalSize);
 
-                    Marshal.ReleaseComObject(propertyStore);
-                    return new Size(width, height);
+                        Marshal.ReleaseComObject(propertyStore);
+                        return new Size(width, height);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine(ex.Message);
                 }
```

Comment:
For several file types including HEIC with no thumbnails, the crash was actually when attempting to get ImageSize. We currently try 2 ways of getting image size, but both throw an exception. In the future, might want to look into yet another way of getting ImageSize for those file types, and catch exceptions along the way so we can try all 3 methods. (I do see file dimensions in File Explorer->Properties->ExtraProperties, so there must be an API that can get it...)